src/test/java/org/ballerinalang/tests/connectors/http/ContentEncodingTests.java

```diff
@@ -0,0 +1,107 @@
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * Test class for testing transfer and content encoding
+ * Uses EncodingService.bal
+ */
+public class ContentEncodingTests extends BallerinaBaseTest {
+
+
+    /**
+     * Tests small chunked message for a Post request
+     */
+    @Test public void chunkedMessageForPostWithSmallPayload() throws IOException {
+
+        HttpTestClient request = new HttpTestClient(ballerinaURL);
+
+        String payload = "This is the payload";
+        request.setChunked(true);
+        request.setContentType("text/plain");
+        ResponseHolder response = request.postRequest("/encoded/spayload", payload);
+
+        assertEquals(response.getResponseMessage(), payload);
+        assertEquals(response.getResponseCode(), 200);
+
+    }
+
+    /**
+     * Tests a large chunked message for a Post request
+     */
+    @Test public void chunkedMessageForPostWithLargePayload() throws IOException {
+
+        HttpTestClient request = new HttpTestClient(ballerinaURL);
+
+        File file = new File("src/test/java/resources/content.txt");
+        FileInputStream inputStream = new FileInputStream(file);
+        InputStreamRequestEntity streamRequestEntity = new InputStreamRequestEntity(inputStream, "text/plain");
+        //String payload = "This is the payload";
+        request.setChunked(true);
+        ResponseHolder response = request.postRequest("/encoded/spayload", inputStream);
+
+        assertNotNull(response.getResponseMessage());
+        assertEquals(response.getResponseCode(), 200);
+
+    }
+
+    @Test public void chunkedGzipEncodedMessageForPost() throws IOException {
+
+        HttpTestClient request = new HttpTestClient(ballerinaURL);
+
+        String payload = "This is the payload";
+        request.setChunked(true);
+        request.setContentEncoded(true);
+        request.setContentType("text/plain");
+        ResponseHolder response = request.postRequest("/encoded/spayload", payload);
+
+        assertEquals(response.getResponseMessage(), payload);
+        assertEquals(response.getResponseCode(), 200);
+
+    }
+
+    @Test public void chunkedGzipEncodedMessageWithAcceptEncodingForPost() throws IOException {
+
+        HttpTestClient request = new HttpTestClient(ballerinaURL);
+
+        String payload = "This is the payloadXXXXXXXXXXXXXXXXXX";
+        request.setChunked(false);
+        request.setContentEncoded(true);
+        request.setHeader("Accept-Encoding", "gzip");
+        request.setContentType("text/plain");
+        ResponseHolder response = request.postRequest("/encoded/spayload", payload);
+
+        GzipCompressor compressor = new GzipCompressor();
+        String decompressedMessage = compressor.decompress(new ByteArrayInputStream(response.getResponseAsByteArray()));
+        assertEquals(decompressedMessage, payload);
+        assertEquals(response.getResponseCode(), 200);
+
+    }
+
+    //    @Test public void chunkedMessageForPut() throws Exception {
```

Comment:
Remove commented code. If a test is disabled temporarily, use the annotation to set it to disabled.

Reply:
This is fixed with #10